sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3730,7 +3728,11 @@ object GpuOverrides extends Logging {
       // This needs to match what murmur3 supports.
       PartChecks(RepeatingParamCheck("hash_key",
         (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 +
-            TypeSig.STRUCT).nested(), TypeSig.all)),
+            TypeSig.STRUCT).nested() +
+            TypeSig.ARRAY.nested(
```

Comment:
I am not an expert on `TypeSig`, it just seems a bit weird that we are calling `nested()` and the `nested(child type)` for `ARRAY`. Why do we need to call both?

Reply:
I am not an expert on TypeSig either and this was my misunderstanding. I thought the `nested` children are kept in separate buckets under parents, but that's not true. We hold the initialTypes and childTypes as flat lists. I will fix this to avoid the confusion.